addons/im_livechat/models/res_users.py

```diff
@@ -11,11 +11,18 @@ class Users(models.Model):
     _inherit = 'res.users'
 
     livechat_username = fields.Char("Livechat Username", help="This username will be used as your name in the livechat channels.")
+    livechat_langs = fields.Many2many(comodel_name='res.lang', string='Livechat languages',
+                            help="These languages, in addition to your main language, will be used to assign you to Live Chat sessions.")
+    has_access_livechat = fields.Boolean(compute='_compute_has_access_livechat', string='Has access to Livechat', store=False, readonly=True)
+
+    def _compute_has_access_livechat(self):
+        for user in self:
+            user.has_access_livechat = user.has_group('im_livechat.im_livechat_group_user') or user.has_group('im_livechat.im_livechat_group_manager')
 
     @property
     def SELF_READABLE_FIELDS(self):
-        return super().SELF_READABLE_FIELDS + ['livechat_username']
+        return super().SELF_READABLE_FIELDS + ['livechat_username', 'livechat_langs']
 
     @property
     def SELF_WRITEABLE_FIELDS(self):
-        return super().SELF_WRITEABLE_FIELDS + ['livechat_username']
+        return super().SELF_WRITEABLE_FIELDS + ['livechat_username', 'livechat_langs']
```

Comment:
To me, fields that are “self-writable“ on `res.users` are usually a symptom that they should rather have been on `res.users.settings`

Reply:
Yeah, indeed. But I think it was because `res.users.settings` was only used by mail. And `SELF_WRITEABLE_FIELDS ` used for all `res.users`

Reply:
`res.users.settings` has been moved to base now 😤

Reply:
Since `livechat_username` is using that already, I will use that in this PR to make it clean and easy. At the same time, I will have another PR for the thing you mentioned and replace all the `SELF_WRITEABLE_FIELDS`  with the `res.users.settings`. This will be more like a tech change and require some changes in other models.

Reply:
https://www.odoo.com/web#menu_id=4720&cids=1&action=333&active_id=1519&model=project.task&view_type=form&id=3491576